test/dotnet-new3.UnitTests/DotnetNewLocaleTests.cs

```diff
@@ -90,5 +90,178 @@ public void TestDotnetCLIEnvVariable(string dotnetCliEnvVar, string expectedName
+  In localization file under the post action with id 'pa1', there are localized strings for manual instruction(s) with ids 'do-not-exist'. These manual instructions do not exist in the template.json file and should be removed from localization file.
+  Post action(s) with id(s) 'pa0' specified in the localization file do not exist in the template.json file. Remove the localized strings from the localization file.",
+
+@$"Warning: Localization file {testTemplateLocation + Path.DirectorySeparatorChar}.template.config/localize/templatestrings.tr.json is not compatible with base configuration {testTemplateLocation + Path.DirectorySeparatorChar}.template.config/template.json, and will be skipped.
+  Post action(s) with id(s) 'pa6' specified in the localization file do not exist in the template.json file. Remove the localized strings from the localization file."
+  };
+
+            var commandResult = new DotnetNewCommand(_log, "-i", testTemplateLocation)
+                .WithCustomHive(home)
+                .WithWorkingDirectory(workingDir)
+                .Execute();
+
+            commandResult.Should()
+                .ExitWith(0)
+                .And
+                .NotHaveStdErr()
+                .And.HaveStdOutContaining($"Success: {testTemplateLocation} installed the following templates:").And.HaveStdOutContaining("TestAssets.Invalid.Localiation.ValidationFailure");
+            foreach (var error in expectedErrors)
+            {
+                commandResult.Should().HaveStdOutContaining(error);
+            }
+        }
+
+        [Fact]
+        public void SkipsLocalizationOnInstantiate_WhenInvalidFormat()
+        {
+            var home = TestUtils.CreateTemporaryFolder("Home");
+            var workingDir = TestUtils.CreateTemporaryFolder();
+            var validTestTemplateLocation = TestUtils.GetTestTemplateLocation("TemplateWithLocalization");
+            var invalidTestTemplateLocation = TestUtils.GetTestTemplateLocation("Invalid/Localization/InvalidFormat");
+            var tmpTemplateLocation = TestUtils.CreateTemporaryFolder();
+            DirectoryCopy(validTestTemplateLocation, tmpTemplateLocation, copySubDirs: true);
+
+            new DotnetNewCommand(_log, "-i", tmpTemplateLocation)
+                .WithCustomHive(home)
+                .WithWorkingDirectory(workingDir)
+                .WithEnvironmentVariable("DOTNET_CLI_UI_LANGUAGE", "de-DE")
+                .Execute()
+                .Should()
+                .ExitWith(0)
+                .And
+                .NotHaveStdErr()
+                .And.NotHaveStdOutContaining("Warning: Failed to read parse localization file").And.NotHaveStdOutContaining("localize/templatestrings.de-DE.json")
+                .And.HaveStdOutContaining($"Success: {tmpTemplateLocation} installed the following templates:").And.HaveStdOutContaining("TemplateWithLocalization")
```

Comment:
Not sure what we can do now, so just as an FYI: this will fail once we have the localizations.
Because we are
- setting the CLI language to German, then
- checking for English strings in the logs.

Same in the other similar cases below.

Reply:
Nice catch. We will need to fix it once localizations are available to use German strings here.